lib/web/integrations_awsoidc_test.go

```diff
@@ -1146,3 +1152,292 @@ func TestAWSOIDCAppAccessAppServerCreationDeletion(t *testing.T) {
+			name:        "services exist for multiple regions",
+			integration: "my-integration",
+			regions:     []string{"us-west-2"},
+			servicesPerRegion: func(t *testing.T) map[string][]*integrationv1.DeployedDatabaseService {
+				command := buildCommandDeployedDatabaseService(t, true, types.Labels{"vpc": []string{"vpc1", "vpc2"}})
+				return map[string][]*integrationv1.DeployedDatabaseService{
+					"us-west-1": dummyDeployedDatabaseServices(1, command),
+					"us-west-2": dummyDeployedDatabaseServices(1, command),
+				}
+			},
+			expectedServices: func(t *testing.T) []ui.AWSOIDCDeployedDatabaseService {
+				return []ui.AWSOIDCDeployedDatabaseService{{
+					Name:                "database-service-vpc-0",
+					DashboardURL:        "url",
+					ValidTeleportConfig: true,
+					MatchingLabels: []libui.Label{
+						{Name: "vpc", Value: "vpc1"},
+						{Name: "vpc", Value: "vpc2"},
+					},
+				}}
+			},
+		},
+		{
+			name:        "service exist but has invalid configuration",
+			integration: "my-integration",
+			regions:     []string{"us-west-2"},
+			servicesPerRegion: func(t *testing.T) map[string][]*integrationv1.DeployedDatabaseService {
+				command := buildCommandDeployedDatabaseService(t, false, nil)
+				return map[string][]*integrationv1.DeployedDatabaseService{
+					"us-west-2": dummyDeployedDatabaseServices(1, command),
+				}
+			},
+			expectedServices: func(t *testing.T) []ui.AWSOIDCDeployedDatabaseService {
+				return []ui.AWSOIDCDeployedDatabaseService{{
+					Name:                "database-service-vpc-0",
+					DashboardURL:        "url",
+					ValidTeleportConfig: false,
+				}}
+			},
+		},
+		{
+			name:        "supports pagination",
+			integration: "my-integration",
+			regions:     []string{"us-west-2"},
+			servicesPerRegion: func(t *testing.T) map[string][]*integrationv1.DeployedDatabaseService {
+				command := buildCommandDeployedDatabaseService(t, true, types.Labels{"vpc": []string{"vpc1", "vpc2"}})
+				return map[string][]*integrationv1.DeployedDatabaseService{
+					"us-west-2": dummyDeployedDatabaseServices(1_024, command),
+				}
+			},
+			expectedServices: func(t *testing.T) []ui.AWSOIDCDeployedDatabaseService {
+				var ret []ui.AWSOIDCDeployedDatabaseService
+				for i := 0; i < 1_024; i++ {
+					ret = append(ret, ui.AWSOIDCDeployedDatabaseService{
+						Name:                fmt.Sprintf("database-service-vpc-%d", i),
+						DashboardURL:        "url",
+						ValidTeleportConfig: true,
+						MatchingLabels: []libui.Label{
+							{Name: "vpc", Value: "vpc1"},
+							{Name: "vpc", Value: "vpc2"},
+						},
+					})
+				}
+				return ret
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			clt := &mockDeployedDatabaseServices{
+				integration:       tt.integration,
+				servicesPerRegion: tt.servicesPerRegion(t),
+			}
+			got, err := listDeployedDatabaseServices(ctx, logger, tt.integration, tt.regions, clt)
+			require.NoError(t, err)
+			require.Equal(t, tt.expectedServices(t), got)
```

Comment:
nit: make the traced line number clear in case either want != got or the expected services getter fails the test

```suggestion
			want := tt.expectedServices(t)
			require.Equal(t, want, got)
```